### PR TITLE
Add nonbonded exclusion classes

### PR DESCRIPTION
**Pull request by victorusu**
I have taken the list of supported interactions from http://manual.gromacs.org/documentation/current/reference-manual/functions/nonbonded-interactions.html

### CODE REVIEW

File: src/gromacs/nblib/exclusions.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * \author Sebastian Keller <keller@cscs.ch>
+ * \author Artem Zhmurov <zhmurov@gmail.com>
+ */
+#ifndef GMX_NBLIB_EXCLUSIONS_H
+#define GMX_NBLIB_EXCLUSIONS_H
+
+namespace nblib
+{
+
+//! Exclusion type
```
Comment (zhmurov) [inline src/gromacs/nblib/exclusions.h]: We use the following doxygen format for multi-line comments:
/*! \brief ...
 *
 */

File: src/gromacs/nblib/exclusions.h [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * derived work must not be called official GROMACS. Details are found
+ * in the README & COPYING files - if they are missing, get the
+ * official version at http://www.gromacs.org.
+ *
+ * To help us fund GROMACS development, we humbly ask that you cite
+ * the research papers on the package. Check out http://www.gromacs.org.
+ */
+/*! \inpublicapi \file
+ * \brief
+ * Implements nblib simulation box
```
Comment (zhmurov) [inline src/gromacs/nblib/exclusions.h]: No it is not

**Review by zhmurov — APPROVED** [review]

File: src/gromacs/nblib/exclusions.h (hunk 1 above)
Comment (victorusu) [reply src/gromacs/nblib/exclusions.h]: Can I use the `//!` like it is described here?
http://www.doxygen.nl/manual/docblocks.html

File: src/gromacs/nblib/exclusions.h [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+{
+
+/*! \brief
+ * Exclusion type
+ * It represents that two particles are excluded for all types of nonbonded interactions
+ */
+class Exclusion
+{
+public:
+    Exclusion() = default;
```
Comment (ejjordan) [inline src/gromacs/nblib/exclusions.h]: This breaks the rule of 5.

File: src/gromacs/nblib/exclusions.h [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+/*! \brief
+ * LJExclusion type
+ * It represents that two particles are excluded for interactions using
+ * the C6/C12 and epsilon/sigma functional forms
+ */
+class LJExclusion
+{
+public:
+    LJExclusion() = default;
```
Comment (ejjordan) [inline src/gromacs/nblib/exclusions.h]: same

File: src/gromacs/nblib/exclusions.h [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+/*! \brief
+ * BuckinghamExclusion type
+ * It represents that two particles are excluded for interactions using
+ * the Buckingham functional form
+ */
+class BuckinghamExclusion
+{
+public:
+    BuckinghamExclusion() = default;
```
Comment (ejjordan) [inline src/gromacs/nblib/exclusions.h]: same

File: src/gromacs/nblib/exclusions.h [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+/*! \brief
+ * CoulombExclusion type
+ * It represents that two particles are excluded for interactions using
+ * the Buckingham functional form
+ */
+class CoulombExclusion
+{
+public:
+    CoulombExclusion() = default;
```
Comment (ejjordan) [inline src/gromacs/nblib/exclusions.h]: same

**Review by ejjordan — CHANGES_REQUESTED** [review]

**Review by ejjordan — APPROVED** [review]

**Pull request merged**